src/extension/infer.rs

```diff
@@ -324,8 +334,12 @@ impl UnificationContext {
             let sig: &OpType = hugr.get_nodetype(tgt_node).into();
             // Incoming ports with a dataflow edge
             for port in hugr.node_inputs(tgt_node).filter(|src_port| {
-                matches!(sig.port_kind(*src_port), Some(EdgeKind::Value(_)))
-                    || matches!(sig.port_kind(*src_port), Some(EdgeKind::Static(_)))
+                matches!(
+                    sig.port_kind(*src_port),
+                    Some(EdgeKind::Value(_))
+                        | Some(EdgeKind::Static(_))
+                        | Some(EdgeKind::ControlFlow)
```

Comment:
I think this neat little one-line change is doing most of the work!?

Which is cool. But can you
1. update the comment two lines above: "Incoming ports with a dataflow edge" doesn't seem right anymore.
2. (super-nit) 10 lines above, "Seperate loop so that...." -> "Separate loop so that". The former is an adjective, the latter a verb...